crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -195,13 +212,72 @@ impl OnSettlementEventUpdater {
 
         tracing::debug!(?hash, ?update, "updating settlement details for tx");
 
-        self.db
-            .update_settlement_details(update.clone())
+        Postgres::update_settlement_details(&mut ex, update.clone())
             .await
             .with_context(|| format!("insert_settlement_details: {update:?}"))?;
-
+        ex.commit().await?;
         Ok(true)
     }
+
+    /// With solver driver colocation solvers are supposed to append the
+    /// `auction_id` to the settlement calldata. This function tries to
+    /// recover that `auction_id`. This function only returns an error if
+    /// retrying the operation makes sense. If all went well and there
+    /// simply is no sensible `auction_id` `Ok(None)` will be returned.
+    async fn recover_auction_id_from_calldata(
+        ex: &mut PgConnection,
+        tx: &Transaction,
+    ) -> Result<Option<i64>> {
+        let tx_from = tx.from.context("tx is missing sender")?;
+        let metadata = match DecodedSettlement::new(&tx.input.0) {
+            Ok(settlement) => settlement.metadata,
+            Err(err) => {
+                tracing::warn!(
+                    ?tx,
+                    ?err,
+                    "could not decode settlement tx, unclear which auction it belongs to"
+                );
+                return Ok(None);
+            }
+        };
+        let auction_id = match metadata {
+            Some(bytes) => i64::from_be_bytes(bytes.0),
+            None => {
+                tracing::warn!(?tx, "could not recover the auction_id from the calldata");
+                return Ok(None);
+            }
+        };
+
+        let score = database::settlement_scores::fetch(ex, auction_id).await?;
+        let data_already_recorded =
+            database::auction_transaction::data_exists(ex, auction_id).await?;
+        match (score, data_already_recorded) {
+            (None, _) => {
+                tracing::debug!(
+                    auction_id,
+                    "calldata claims to settle auction that has no competition"
+                );
```

Comment:
nit: This case will be entered for each transaction exactly once on different environment (will be entered in prod for staging transaction).
Nice to have for completness but maybe worth skipping since does not tell us much.